days/src/day2/Day2.kt

```diff
@@ -0,0 +1,43 @@
+package day2
+
+import Day
+import java.io.File
+
+interface Policy {
+    fun isValid(password: String): Boolean
+}
+data class PolicyPuzzle1(val low: Int, val high: Int, val symbol: Char): Policy {
+    override fun isValid(password: String) = password.count { it == symbol } in low..high
+}
+
+data class PolicyPuzzle2(val index1: Int, val index2: Int, val symbol: Char): Policy {
+    override fun isValid(password: String) = (password[index1-1] == symbol || password[index2-1] == symbol) &&
+                    password[index1-1] != password[index2-1]
```

Comment:
I truncated this to `((password[index1-1] == symbol) != (password[index2-1] == symbol))`, although it looks like Kotlin has [a logical XOR operator](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/xor.html)